### PR TITLE
[Merged by Bors] - feat(CategoryTheory): translate between Yoneda and "relative Yoneda"

### PR DESCRIPTION
**Pull request by TwoFX**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Comma/Presheaf.lean [hunk 1]
```diff
@@ -572,4 +573,34 @@ def CostructuredArrow.toOverCompOverEquivPresheafCostructuredArrow (A : Cᵒᵖ
     CostructuredArrow.toOver yoneda A ⋙ (overEquivPresheafCostructuredArrow A).functor ≅ yoneda :=
   toOverYonedaCompRestrictedYoneda A
 
+/-- This isomorphism says that hom-sets in the category `Over A` for a presheaf `A` where the domain
+    is of the form `(CostructuredArrow.toOver yoneda A).obj X` can instead be interpreted as
+    hom-sets in the category `(CostructuredArrow yoneda A)ᵒᵖ ⥤ Type v` where the domain is of the
+    form `yoneda.obj X` after adjusting the codomain accordingly. This is desirable because in the
+    latter case the Yoneda lemma can be applied. -/
+def CostructuredArrow.toOverCompYoneda (A : Cᵒᵖ ⥤ Type v) (T : Over A) :
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Comma/Presheaf.lean:581]: Should these be tagged with `@[simps]`? I think they need some sort of simp API around them...

**Review by jcommelin — COMMENTED** [review]
awaiting-author

File: Mathlib/CategoryTheory/Comma/Presheaf.lean (hunk 1 above)
Comment (TwoFX) [reply Mathlib/CategoryTheory/Comma/Presheaf.lean:586]: Thanks for bringing this up. In my application #17451, only the abstract existence of the natural isomorphism is required, but I agree that it should be possible to look into these definitions (unlike `overEquivPresheafCostructuredArrow` and `CostructuredArrow.toOverCompOverEquivPresheafCostructuredArrow` which should be black boxes). The `simps` lemmas didn't seem particularly great, so I added some API lemmas which look useful when thinking about the results using pen and paper.

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**